src/api/projects/{id}/get.ts

```diff
@@ -0,0 +1,69 @@
+import { Request, Response } from "express";
+import User from "../../../models/User";
+import Project, { DbToStdModal_Project } from "../../../models/Project";
+import { IProject } from "../../../models/types";
+import { genericServerError, validateAuthenticationHeader } from "../../../common/helpers/generic";
+import { GetDiscordIdFromToken } from "../../../common/helpers/discord";
+import { HttpStatus, BuildResponse, ResponsePromiseReject } from "../../../common/helpers/responseHelper";
+
+module.exports = async (req: Request, res: Response) => {
+    let id = req.params['id'];
+
+    // If someone wants the projects for a specific user, they must be authorized
+    const authAccess = validateAuthenticationHeader(req, res);
+    if (!authAccess) return;
+
+    const authenticatedDiscordId = await GetDiscordIdFromToken(authAccess, res);
+
+    if (authenticatedDiscordId) {
+        const result = await getPrjectById(id).catch(err => genericServerError(err, res));
```

Comment:
To format the response object correctly, use
```catch(err => genericServerError(err, res));```
in place of
```catch((err: IRequestPromiseReject) => BuildResponse(res, err.status, err.reason))```

Reply:
maybe you inverted last 2 comments, if I understood correctly should be fixed now